packages/block-scripts/shared/ensure-block-schema-in-dist.js

```diff
@@ -0,0 +1,38 @@
+import * as TJS from "typescript-json-schema";
+import fs from "fs-extra";
+import path from "path";
+import { blockDistDirPath, blockRootDirPath } from "./paths.js";
+
+export const ensureBlockSchemaInDist = async () => {
+  const blockSchemaInDistFilePath = path.resolve(
+    blockDistDirPath,
+    "block-schema.json",
+  );
+
+  const customBlockSchemaFilePath = path.resolve(
+    blockRootDirPath,
+    "block-schema.json",
+  );
+
+  if (await fs.pathExists(customBlockSchemaFilePath)) {
+    await fs.copy(customBlockSchemaFilePath, blockSchemaInDistFilePath);
+    console.log("Using custom block-schema.json");
+    return;
+  }
+
+  try {
+    const tsconfigFilePath = path.resolve(blockRootDirPath, "tsconfig.json");
+    const program = TJS.programFromConfig(tsconfigFilePath);
+
+    /** @type {import("typescript-json-schema").PartialArgs} */
+    const settings = {
+      required: true,
+    };
+
+    const schema = TJS.generateSchema(program, "AppProps", settings);
+    await fs.ensureDir(blockDistDirPath);
+    await fs.writeJson(blockSchemaInDistFilePath, schema, { spaces: 2 });
+  } catch (error) {
+    console.warn(`Unable to generate block-schema: ${error}`);
```

Comment:
Maybe this should fallback to generating an empty schema? i.e. `{}`. In case the block accepts no props. And still warn about it – then at least the file is there for loading into HASH.

Reply:
👍 242f74454a0a3755a87de92b3fc78646cf2b2873